files/en-us/web/javascript/reference/global_objects/object/hasown/index.html

```diff
@@ -0,0 +1,153 @@
+
+<p>It is recommended over {{jsxref("Object.hasOwnProperty()")}} because it works for objects created using <code>Object.create(null)</code> and with objects that have overridden the inherited <code>hasOwnProperty()</code> method.
+  While it is possible to workaround these problems by calling <code>Object.prototype.hasOwnProperty()</code> on an external object, <code>Object.hasOwn()</code> is more intuitive.</p>
+ 
+<h2 id="Examples">Examples</h2>
+
+<h3>Using hasOwn to test for a property's existence</h3>
+
+<p>The following example shows how to test whether an object directly defines ('owns') a particular property:</p>
+
+<pre class="brush: js">let object1 = {};
+object1.prop = 'exists';
+
+// Normal way to use the method
+if (Object.hasOwn(object1, 'prop')) {
+  // true - 'prop' is defined
+}
+  
+// Own property values can be null or undefined
+object1.prop_value_null = null;
+Object.hasOwn(object1, 'prop_value_null');  // true
+object1.prop_value_undefined = undefined;
+Object.hasOwn(object1, 'prop_value_undefined');  // true
+  
+// Inherited and undeclared properties return false
+Object.hasOwn(object1,'toString');  // false
+Object.hasOwn(object1,'prop_not_defined');  // false
+</pre>
+
+<h3 id="Iterating_over_the_properties_of_an_object">Iterating over the properties of an object</h3>
+
+<p>The following example shows how to iterate over the enumerable properties of an object without executing on inherited properties.</p>
+
+<pre class="brush: js">let buz = {
+  fog: 'stack'
+};
+
+for (let name in buz) {
+  if (Object.hasOwn(buz, name)) {
+    console.log('this is fog (' +
+      name + ') for sure. Value: ' + buz[name]);
+  }
+  else {
+    console.log(name); // toString or something else
+  }
+}
+</pre>
+
+<p>Note that the {{jsxref("Statements/for...in", "for...in")}} loop only iterates enumerable items: the absence of non-enumerable properties emitted from the loop does not imply that 
+  <code>hasOwn</code> itself is confined strictly to enumerable items (as with {{jsxref("Object.getOwnPropertyNames()")}}).</p>
+
+<h3 id="checking_if_an_array_index_exists">Checking if an Array index exists</h3>
+
+  <p>The elements of an {{jsxref("Array")}} are defined as direct properties, so you can use <code>hasOwn()</code> method to check whether a particular index exists:</p>
+    
+    <pre class="brush: js">let fruits = ['Apple', 'Banana','Watermelon', 'Orange'];
+    Object.hasOwn(fruits, 3);   // true ('Orange')
+    Object.hasOwn(fruits, 4);   // false - not defined
+    </pre>
+
+<h3 id="problematic_cases_for_hasownproperty">Problematic cases for hasOwnProperty</h3>
+
+<p>This section demonstrate that <code>hasOwn()</code> is immune to the problems that affect <code>hasOwnProperty</code>.
+  Firstly, it can be used with objects that have reimplemented <code>hasOwnProperty()</code>:</p>
+
+<pre class="brush: js">let foo = {
+    hasOwnProperty: function() {
```

Comment:
Indentation is a bit messed up

```suggestion
  hasOwnProperty: function() {
```